starport_template/lib/pages/transfer_asset_page.dart

```diff
@@ -0,0 +1,125 @@
+import 'package:cosmos_ui_components/cosmos_ui_components.dart';
+import 'package:flutter/material.dart';
+import 'package:starport_template/entities/balance.dart';
+import 'package:starport_template/entities/transaction.dart';
+import 'package:starport_template/pages/sign_transaction_page.dart';
+import 'package:starport_template/widgets/balance_card_list.dart';
+
+class TransferAssetPage extends StatefulWidget {
+  final Balance balance;
+
+  const TransferAssetPage({Key? key, required this.balance}) : super(key: key);
+
+  @override
+  State<TransferAssetPage> createState() => _TransferAssetPageState();
+}
+
+class _TransferAssetPageState extends State<TransferAssetPage> {
+  double amount = 0.0;
+  double fee = 0.0;
+  String walletAddress = '';
+
+  bool get isTransferValidated => amount != 0.0 && walletAddress.isNotEmpty && fee != 0.0;
+
+  @override
+  Widget build(BuildContext context) {
+    final theme = CosmosTheme.of(context);
+    return Scaffold(
+      appBar: CosmosAppBar(
+        leading: const CosmosBackButton(),
+        title: 'Transfer ${widget.balance.denom.text}',
+      ),
+      body: Column(
+        children: [
+          SizedBox(height: theme.spacingXXXL),
+          CosmosBalanceCard(
+            denomText: widget.balance.denom.text,
+            amountDisplayText: widget.balance.amount.displayText,
+            isListTileType: true,
+          ),
+          SizedBox(height: theme.spacingXXL),
+          _buildTextFields(theme, context),
+          const Spacer(),
+          _buildFooterButton(theme),
```

Comment:
no need to prefix the method with `_build`. simple `_footerButton(theme)` is enough as it indicates the method creates a button

Reply:
Updated